### PR TITLE
Toggles for testing lot 1

### PR DESCRIPTION
**Pull request by Yelia41**

### CODE REVIEW

File: app/manifest.json [hunk 1]
```diff
@@ -42,15 +42,7 @@
       "run_at": "document_idle"
     }
   ],
-  "applications": {
```
Comment (abarre) [inline app/manifest.json:45]: This code is used for firefox compatibility. Don't remove it.

File: app/manifest.json [hunk 2]
```diff
@@ -42,15 +42,7 @@
...
-    "gecko": {
-      "id": "{c1687a9a-9054-430e-94cf-2ef9b3caeb7b}",
-      "update_url": "https://raw.githubusercontent.com/fasterize/fasterize-web-extension/master/app/update-manifest.json",
-      "strict_min_version": "48.0"
-    }
-  },
   "web_accessible_resources": [
     "show_deferjs_trace.js"
-  ],
-  "incognito": "split"
```
Comment (abarre) [inline app/manifest.json:55]: This code is used for incognito mode. Don't remove it.

File: app/popup/popup.html [hunk 3]
```diff
@@ -203,6 +204,61 @@ <h3><strong id="statusCode"></strong></h3>
               Disable trace
             </button>
           </div>
+          <div class="tab-pane" id="testscookies">
+
+            <table width="100%" cellspacing="1" cellpadding="5" id="parent-list">
```
Comment (abarre) [inline app/popup/popup.html:209]: I think that `parent-list` is not explicit. Can you rename it `feature-flag-table` ?

File: app/popup/popup.js [hunk 4]
```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
       request.showLazyloadedImages();
     });
 
+    document.getElementById("parent-list").addEventListener("click", function (e) {
+      if (e.target && e.target.nodeName == "INPUT") {
+        jeclique(e.target.id);
+      }
+    });
+    
+    function jeclique(cookieName) {
```
Comment (abarre) [inline app/popup/popup.js:203]: Could you rename this method with a more explicit name. I suggest `function toggleFlag(featureFlagName) {`

File: app/popup/popup.js [hunk 5]
```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
       request.showLazyloadedImages();
     });
 
+    document.getElementById("parent-list").addEventListener("click", function (e) {
```
Comment (abarre) [inline app/popup/popup.js:197]: Could you change `parent-list` ? Cf my previous comment.

File: app/popup/popup.js [hunk 6]
```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
...
+      if (e.target && e.target.nodeName == "INPUT") {
+        jeclique(e.target.id);
+      }
+    });
+    
+    function jeclique(cookieName) {
+      var toggle = true;
+      toggle = document.getElementById(cookieName).checked;
+      if (toggle == false) {
```
Comment (abarre) [inline app/popup/popup.js:206]:
```suggestion
      if (!toggle) {
```

File: app/popup/popup.js [hunk 7]
```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
...
+        browser.tabs.query({ active: true, lastFocusedWindow: true }).then(tabs => {
```
Comment (abarre) [inline app/popup/popup.js:207]: The code is duplicated. The only change is `m_dict = this.getParamsDict(cookieName, false | true)`

File: app/popup/popup.js [hunk 8]
```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
...
+        url += `${key}=${value}&`;
+      }
+      if (url.length > 1) {
+        url = url.substr(0, url.length - 1);
+      }
+      return url;
+    }
+
+    getParamsDict = (p_string, p_value) => {
+      if (p_value == false || p_value == true) {
```
Comment (abarre) [inline app/popup/popup.js:255]: this method is duplicated. The only change is `p_dict[param[0]] = p_value` and `p_dict[param[0]] = param[1]`

File: app/popup/popup.js [hunk 9]
```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
...
+        for (let i = 0; i < params.length; i++) {
+          let param = params[i].split("=")
+          p_dict[param[0]] = param[1]
+        }
+        return p_dict;
+      }
+    }
+
+    document.getElementById("ongletCookies").addEventListener("click", function (e) {
+	//$("#ongletCookies").click(function () {
```
Comment (abarre) [inline app/popup/popup.js:276]: remove this comment

File: app/popup/popup.js [hunk 10]
```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
...
+      afficherOnglet();
```
Comment (abarre) [inline app/popup/popup.js:277]: use an english word `showTab`

**Review by abarre — CHANGES_REQUESTED** [review]

File: app/manifest.json (hunk 1 above)
Comment (Yelia41) [reply app/manifest.json:45]: Ok mais cela crée des erreurs sous Chrome

File: app/manifest.json (hunk 2 above)
Comment (Yelia41) [reply app/manifest.json:55]: Ok mais cela crée des erreurs sous Firefox...

File: app/manifest.json (hunk 1 above)
Comment (Yelia41) [reply app/manifest.json]: J'ai corrigé tous tes commentaires, tiens moi au courant ;) Merci!

**Review by abarre — APPROVED** [review]

File: app/manifest.json (hunk 2 above)
Comment (abarre) [reply app/manifest.json]: cette clé est supprimée pour firefox lors de la release.

**Pull request merged**